Week3/homework/js-exercises/creditCardValidator.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+{
+  // In this project you'll write a script
+  // that validates whether or not a credit card number is valid.
+  // Here are the rules for a valid number:
+  function checkCard(cardNumber) {
+
+    // Number must be 16 digits
+    if (cardNumber.length !== 16) {
+      console.log(`Card must be 16 digits!`);
+      return false;
+    }
+
+    // All of them must be numbers
+    for (let item = 0; item < cardNumber.length; ++item) {
+      let cardNumberItem = cardNumber[item];
+      cardNumberItem = Number.parseInt(cardNumberItem);
+
+      if (Number.isInteger(cardNumberItem) !== true) {
+        console.log(`Must be numbers all of them!`)
+        return false;
+      }
+    }
+
+    // You must have at least two different digits represented
+    // (all of the digits cannot be the same)
+    const object = {};
+    let index = 0;
+    while (index < cardNumber.length) {
+      object[cardNumber[index]] = true;
+      ++index;
+    }
+    if (Object.keys(object).length < 2) {
+      console.log(`It must have at least two diffrent digits!`)
+      return false;
+    }
```

Comment:
Aahaan. I would suggest that you explore a new feature in JavaScript called **Sets**(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) and see if you can use it for achieving the same result but with lesser code 🙂